main.go

```diff
@@ -113,6 +159,14 @@ type API struct {
 	client storepb.StoreClient
 }
 
+func loggerWrap(f func(w http.ResponseWriter, r *http.Request) error, logger log.Logger) func(w http.ResponseWriter, r *http.Request) error {
+	return func(w http.ResponseWriter, r *http.Request) error {
+		ctx := context.WithValue(r.Context(), "logger", logger)
```

Comment:
I'm not sure we need all this complexity here. Potentially you can change `remoteRead` to take a logger and make loggerWrap just pass that through, e.g.:

```
func loggerWrap(f func(w http.ResponseWriter, r *http.Request, logger log.Logger) error, logger log.Logger) func(w http.ResponseWriter, r *http.Request) error {
	return func(w http.ResponseWriter, r *http.Request) error {
	    return f(w, r, logger)
       }
}
```

Reply:
@dgl PTAL